src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -134,49 +91,71 @@ def test_WINNF_FT_S_FAD_1(self):
+            # Get json data from url
+            data = self._sas.DownloadFile(activity_dump_file['url'])
+            if activity_dump_file['recordType'] == 'cbsd':
+                 # Verify that everything in the full dump matches a cbsd or grant created at the beginning
+                for record in data['recordData']:
+                # Verify the response files with CbsdData.schema.json Object schema
+                    self.assertContainsRequiredFields("CbsdData.schema.json", record)
+                    self.assertTrue(record['registration']['fccId'] in (device_a['fccId'], device_c['fccId']))
+                # Verify all the previous activities on CBSDs and Grants exist in the dump files    
+                for index, device in enumerate([device_a, device_c]):
+                    record_id = 'cbsd/'+ device['fccId']+'/'+ hashlib.sha1(device['cbsdSerialNumber']).hexdigest()
+                    cbsd_record = [record['registration'] for record in data['recordData'] if record['id'] == record_id]
+                    self.assertEqual(first_cbsd['cbsdCategory'],second_cbsd['cbsdCategory'])
+                    self.assertEqual(device['airInterface']['radioTechnology'],\
+                                     cbsd_record['airInterface']['radioTechnology'])
+                    self.assertEqual(device['installationParam']['latitude'],\
+                                     cbsd_record['installationParam']['latitude'])
+                    self.assertEqual(device['installationParam']['longitude'],\
+                                     cbsd_record['installationParam']['longitude'])
+                    self.assertEqual(device['installationParam']['height'],\
+                                     cbsd_record['installationParam']['height'])
+                    self.assertEqual(device['installationParam']['heightType'], \
+                                     cbsd_record['installationParam']['heightType'])
+                    self.assertEqual(device['installationParam']['indoorDeployment'], \
+                                     cbsd_record['installationParam']['indoorDeployment'])
+                    self.assertEqual(device['installationParam']['antennaGain'], \
+                                     cbsd_record['installationParam']['antennaGain'])                           
+                    # Get grants by cbsd_id
+                    grants_of_cbsd = [cbsd['grants'] for cbsd in data['recordData'] if cbsd['id'] == record_id]
+                    self.assertEqual(grants_of_cbsd[0][0]['id'], (grant_ids[index]))
+                    self.assertGrantRecord(record['grants'][0], grants_to_request[index], grant_response[index])
+                    # Verify the Grant Of the Cbsd
+                    self.assertEqual(record['grants'][0]['id'],grant_response[index]['grantId'])
+                    if record['grants'][0]['operationParam'] != null :
```

Comment:
I think this is a required object so it can't be 'null'. (Also: Python doesn't have 'null' so this code won't run.)

Reply:
thank you, I changed the null keyword, but for this condition we see in the SAS-SAS spec in the "requestedOperationParam": 
If only this field is present (and not the operationParam field), there is no active grant associated with the request.

so I understood that operation parameters are not presented if the grant is not authorized(in case of suspended for example), so the operation parameters are not always presented, but requestedOperationParam are required always, do you agree with that?

Reply:
My understanding is that `operationParam` is required while `requestedOperationParam` is optional -- for *all* grants, even suspended ones. `operationParam` must match the values given to the CBSD in grantResponse while `requestedOperationParam` signals what EIRP the device *really* wanted (and it is this value which will be used in the IAP and DPA move list calculations).

We have defined an upper bound on the max EIRP in `requestedOperationParam` (the EIRP capability of the CBSD) but exactly how to choose the value is not mandated.

In general, you cannot not know from SAS-SAS exchange whether a grant is suspended or in use.